client/gutenberg/extensions/slideshow/component.js

```diff
@@ -0,0 +1,154 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { createRef, Children, Component } from '@wordpress/element';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+
+import Swiper from 'swiper';
+import 'swiper/dist/css/swiper.min.css';
+
+export class Slideshow extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			images: [],
+			imageHeight: 400,
+		};
+		this.slideshowRef = createRef();
+		this.btnNextRef = createRef();
+		this.btnPrevRef = createRef();
+		this.paginationRef = createRef();
+	}
+	render() {
+		const { children, className } = this.props;
+		const { imageHeight } = this.state;
+		const classNames = classnames( className, 'swiper-container' );
+		return (
+			<div className={ classNames } ref={ this.slideshowRef }>
+				<div className="swiper-wrapper">
+					{ Children.map( children, child => {
+						if ( ! child.props.children ) {
+							return null;
+						}
+						const image = this.getChildByDataAttribute( child, 'data-is-image' );
```

Comment:
These `data-is-*` attributes are a bit redundant... Can we have a method directly look for a given element type instead?

Reply:
I hit a wall on this problem, and the solution you are seeing is effectively the hack I came up with. I was not able to figure out how to derive the tag name from a React component, so the `data-is-*` components serve only as a way to declare the element so the component can properly identify them. This is also the reason why the function is called `getChildByDataAttribute` - ideally it _would_ be called `getChildByPropName` because ideally it would do just that 😄.  If you have thoughts on a truer approach, I would love to try.